### PR TITLE
InvokeFirstWhere

### PR DESCRIPTION
**Pull request by mikaelm12**
In the `DefaultHubLifeTimeManagers ` `InvokeUserAsync` we use `InvokeAllWhere `which scans every connnection every time even though we only care about one. `InvokeFirstWhere `stops after we find the connection we want.

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs [hunk 1]
```diff
@@ -176,7 +195,7 @@ private InvocationMessage CreateInvocationMessage(string methodName, object[] ar
 
         public override Task InvokeUserAsync(string userId, string methodName, object[] args)
         {
-            return InvokeAllWhere(methodName, args, connection =>
+            return InvokeFirstWhere(methodName, args, connection =>
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs]: I don't think we want this. You can have the same user connected from multiple locations and with this change only one would get the message, but all of them should get the message.

Comment (mikaelm12) [reply src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs]: Hmmm, good point. Didn't think of that.

Comment (mikaelm12) [reply src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs]: Should there be a better way to may User Identifies to connection ids though?

**Comment by davidfowl** [comment]
What's the scenario for this?

**Comment by mikaelm12** [comment]
It was to avoid using InvokeAllWhere to implement InvokeUserAsync but as @BrennanConroy pointed out we actually should be using InvokeAllWhere

**Pull request closed without merging**